### PR TITLE
Fix descartes processing of results to handle freespace correctly

### PR DESCRIPTION
**Pull request by Levi-Armstrong**
I also renamed the interpolate function that handled waypoints version because I was getting issues with clang-tidy.

### CODE REVIEW

File: tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/descartes/impl/descartes_motion_planner.hpp [hunk 1]
```diff
@@ -118,25 +118,66 @@ tesseract_common::StatusCode DescartesMotionPlanner<FloatType>::solve(const Plan
...
     {
-      // This instruction corresponds to a composite. Set all results in that composite to the results
-      auto* move_instructions = instruction.get().cast<CompositeInstruction>();
-      for (auto& instruction : *move_instructions)
+      const auto* plan_instruction = instructions_flattened.at(plan_index).get().cast_const<PlanInstruction>();
+
+      if (plan_instruction->isLinear())
+      {
+        // This instruction corresponds to a composite. Set all results in that composite to the results
+        auto* move_instructions = results_flattened[plan_index].get().cast<CompositeInstruction>();
```
Comment (mpowelson) [inline tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/descartes/impl/descartes_motion_planner.hpp:137]: I know you didn't add this, but should we add an assert checking this?

File: tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/descartes/impl/descartes_motion_planner.hpp [hunk 2]
```diff
@@ -118,25 +118,66 @@ tesseract_common::StatusCode DescartesMotionPlanner<FloatType>::solve(const Plan
...
+
+        if (!first_plan_instruction_found)
+        {
+          Eigen::MatrixXd temp = interpolate(start.template cast<double>(),
+                                             stop.template cast<double>(),
+                                             static_cast<int>(move_instructions->size()) - 1);
+
+          assert(temp.cols() == static_cast<long>(move_instructions->size()));
+          for (std::size_t i = 0; i < move_instructions->size(); ++i)
+            (*move_instructions)[i].cast<MoveInstruction>()->setPosition(temp.col(static_cast<long>(i)));
```
Comment (mpowelson) [inline tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/descartes/impl/descartes_motion_planner.hpp:163]: So this is still assuming we don't fill out the start waypoint in the response composites?

File: tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/descartes/impl/descartes_motion_planner.hpp [hunk 3]
```diff
@@ -118,25 +118,66 @@ tesseract_common::StatusCode DescartesMotionPlanner<FloatType>::solve(const Plan
...
+                                                                             dof);
+          instruction.cast<MoveInstruction>()->setPosition(temp.template cast<double>());
+        }
+      }
+      else if (plan_instruction->isFreespace())
       {
-        Eigen::Map<Eigen::Matrix<FloatType, -1, 1>> temp(solution.data() + dof * result_index, dof);
-        instruction.cast<MoveInstruction>()->setPosition(temp.template cast<double>());
-        result_index++;
+        // Because descartes does not support freespace it just includes the plan instruction waypoint so we will
```
Comment (mpowelson) [inline tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/descartes/impl/descartes_motion_planner.hpp:147]: I think this comment may be incorrect? This is including a MoveInstruction

**Comment by Levi-Armstrong** [comment]
replaced by #335

**Pull request closed without merging**